### PR TITLE
Update hero section styles according to BEM methodology

### LINKED ISSUE #1102: Convert homepage hero section styles to use BEM methodology
### Description
The homepage hero section currently does not follow the BEM (Block Element Modifier) methodology that has been partially adopted in newer pages, such as the thanks page. The upcoming navbar and search modal will also follow BEM conventions. The hero section was initially implemented in PR #459, but now needs to be updated to align with these conventions.

### Task  
Update the CSS and structure of the homepage hero section to match the BEM conventions used elsewhere. Only the relevant portions of `index.php` and `styles/home.css` should be updated. Additionally, the visual tests references to the `hero` classes will need to be updated.

### Example Conversion  
Here’s an example to guide you in converting a class:

**Current class:**
```css
.hero-btn-primary { ... }
```

**Converted to BEM:**
```css
.hero__btn--primary { ... }
```

In this example:
- `.hero` is the **Block**, representing the main component.
- `__btn` is an **Element**, indicating a part of the block.
- `--primary` is a **Modifier**, signifying a variation of the button.

### Steps
1. Clone the repository and navigate to the homepage code.
2. Refactor the CSS classes in the hero section using the BEM naming convention. Focus only on the hero-related classes.
3. Ensure the CSS in `styles/home.css` follows BEM principles.
4. Update the relevant visual tests where `hero` classes are used.
5. Test the changes to ensure the styling and visual tests remain consistent.

### Resources
- [BEM methodology documentation](https://getbem.com/introduction/)
- [Existing styles for the PHP 8 release pages](https://github.com/php/web-php/blob/8e82d8d92c123986503e2b3061528077ccfb8c2c/styles/php8.css)
- [Existing styles for the thanks page](https://github.com/php/web-php/blob/master/styles/theme-medium.css#L521-L611)
- [Upcoming navbar and search modal styles](https://github.com/php/web-php/blob/eb7a39d86eae7b1d608bbb7def3d84e50255b40a/styles/theme-medium.css#L187-L723)

### Good first issue for
- Developers interested in learning and applying CSS methodologies like BEM.
- Hacktoberfest participants looking for beginner-friendly contributions.

Feel free to ask questions if you need any help!

### PR DESCRIPTION
**Pull request by nafisreza**
fixes #1102

## Changes:
- All the hero section classes have been updated according to BEM.
- Respective HTML classes have also been updated.
- Visual test has been updated with new class name.

### CODE REVIEW

File: styles/home.css [hunk 1]
```diff
@@ -85,41 +85,41 @@
...
 
-.hero-versions a:hover,
-.hero-versions a:focus {
+.hero__versions a:hover,
+.hero__versions a:focus {
   border-bottom:1px dotted;
 }
 
-.hero-version-link {
+.hero__version--link {
```
Comment (lhsazevedo) [inline styles/home.css:107]: I think version link should be an element: `.hero__version-link`

**Comment by lhsazevedo** [comment]
Everything looks great to me, just a minor suggestion (nit) in the comment I added. Once that's addressed, I think it should be good to go!

File: styles/home.css (hunk 1 above)
Comment (nafisreza) [reply styles/home.css]: Fixed

**Comment by nafisreza** [comment]
Thanks @lhsazevedo. I accidentally overlooked it. 
Fixed in the last commit

**Comment by lhsazevedo** [comment]
LGTM

---

**Note for the person merging:** Please either add the `hacktoberfest-accepted` label to this PR, or follow the suggestion in #1106 to add the `hacktoberfest` label to the repository.

**Review by sy-records — APPROVED** [review]
Thanks!

**Comment by lhsazevedo** [comment]
Thanks for your enthusiasm! The rest of the website could indeed be refactored to follow BEM, which would help a lot with readability and maintenance. However, there's quite a bit of technical debt in the styles that goes beyond just the naming convention. We're dealing with some older techniques, like float-based layouts with clearfixes, which could definitely use a modern overhaul.

Switching to Flexbox (and potentially Grid where it makes sense) would significantly improve the layout's flexibility and remove a lot of that legacy CSS. Converting to BEM alongside these structural updates would make everything cleaner and more modular.

Appreciate the offer to help! It might be best to tackle it one section or block at a time, updating both the structure and the naming conventions incrementally.

Thanks again!

**Pull request merged**